### PR TITLE
Add weighted loss (WIP)

### PR DESCRIPTION
**Pull request by kartikdutt18**
This arises out of #2345. Need to make some changes.
NOTE: This hasn't been tested.

### CODE REVIEW

File: src/mlpack/methods/ann/loss_functions/cross_entropy_error_impl.hpp [hunk 1]
```diff
@@ -32,8 +35,19 @@ CrossEntropyError<InputDataType, OutputDataType>::Forward(
...
+  // suggestions are welcome (Without using for loops). Will figure it out tomorrow then.
+  if (numClasses)
+  {
+    for (size_t i = 0; i < input.n_elem; i++)
+    {
+      error(i) *= weights(target(i));
+    }
+  }
+  return arma::accu(error);
 }
```
Comment (mrityunjay-tripathi) [inline src/mlpack/methods/ann/loss_functions/cross_entropy_error_impl.hpp:51]: Is the minus sign missing here or this is intentional change?

Comment (kartikdutt18) [reply src/mlpack/methods/ann/loss_functions/cross_entropy_error_impl.hpp:51]: Yep missed it by mistake. I haven't tested it yet that's why I opened a draft PR.

**Comment by kartikdutt18** [comment]
Keep open please.

**Comment by kartikdutt18** [comment]
Hey @iamshnoo, I guess you'll be working on this during the summer. So I guess I can close this right?

**Comment by iamshnoo** [comment]
My primary goal is to complete the reduction facilities. But that is going to take some time because some of the functions are incorrectly implemented.
Once that is done, I will add weights facility to some of the loss functions during the summer. Probably a few will be left to complete after the summer.

Anyway, I think you can go ahead and close this one :)

**Comment by kartikdutt18** [comment]
Great. Closing this then.
